python/ray/serve/handle.py

```diff
@@ -574,12 +528,16 @@ async def _to_object_ref(self, _record_telemetry: bool = True) -> ray.ObjectRef:
         assigned to a replica actor. If there are many requests in flight and all
         replicas' queues are full, this may be a slow operation.
         """
-        return await self._to_object_ref_or_gen(_record_telemetry=_record_telemetry)
+
+        ServeUsageTag.DEPLOYMENT_HANDLE_TO_OBJECT_REF_API_USED.record("1")
```

Comment:
Should telemetry be gated by `_record_telemetry`?

Reply:
I think `_record_telemetry` was only used because normal fetching operations like `.result()` was going through `_to_object_ref`, which is a developer API that we normally want to collect telemetry on. With this refactor `_to_object_ref` is not used internally anymore so I don't think `record_telemetry` is needed anymore.